src/date-navigator/date-navigator.tsx

```diff
@@ -18,21 +20,32 @@ export const DateNavigator = ({
     minDate,
     maxDate,
     loading,
+    showDateAsShortForm,
+    showTodayAsToday,
     onLeftArrowClick,
     onRightArrowClick,
+    onCalendarDateSelect,
     ...otherProps
 }: DateNavigatorProps) => {
     // =============================================================================
     // CONST, STATE, REF
     // =============================================================================
     const date = DateHelper.toDayjs(selectedDate);
     const dateText = DateHelper.toDayjs(selectedDate)
-        .format("D MMMM YYYY")
+        .format(showDateAsShortForm ? "D MMM YYYY" : "D MMMM YYYY")
         .toString();
     const isToday = DateHelper.isSame(selectedDate, dayjs());
-    const dayText = isToday
-        ? "Today"
-        : DateHelper.toDayjs(selectedDate).format("dddd");
+    const dayText =
+        isToday && showTodayAsToday
+            ? "Today"
+            :
+            DateHelper.toDayjs(selectedDate).format(showDateAsShortForm ? "ddd" : "dddd");
+    const containerRef = useRef<HTMLDivElement>(null);
+    const { width: referenceWidth } = useResizeDetector({
+        targetRef: containerRef,
+        handleHeight: false,
+    });
```

Comment:
the ElementWithDropdown also provides `elementWidth` for the dropdown, does that work for you? then there's no need to separately handle resizing (refer to renderCalendar in `date-input.tsx`)